packages/aws-cdk-lib/aws-s3-deployment/lib/source.ts

```diff
@@ -224,7 +225,15 @@ export class Source {
     }
     return {
       bind: (scope: Construct, context?: DeploymentSourceContext) => {
-        return Source.data(objectKey, Stack.of(scope).toJsonString(obj), markersConfig).bind(scope, context);
+        // toJsonString can generate extra tokens that would not be needed if the obj is not a token
```

Comment:
Is it feasible to fix `toJsonString`/`toYamlString` to deal with this instead? If these functions are used in other places in the future, they might need conditional checks as well.

This is not a blocker, since getting the issue resolved is more important. If there is no straightforward path to fix the functions directly, let's add a backlog issue.

Reply:
`toJsonString` and `toYamlString` is used by different functions across CDK, so changing it has a high impact. The functions' behavior isn't a bug as they produce the correct values once resolved. Its just that within the context of our fix, we'd like to not generate extra tokens as we's like to process the underlying tokens directly to eventually replace them with markers.